### PR TITLE
CONSOLE-3684: Convert components in modals folder from class componen…

### PR DESCRIPTION
**Pull request by tvu20**
…t to functional component

The following files in frontend/public/components/modals have been converted from a class component to a functional component in order to use react-router v6 hooks:

add-secret-to-workload.tsx
create-namespace-modal.jsx

### CODE REVIEW

File: frontend/public/components/modals/add-secret-to-workload.tsx [hunk 1]
```diff
@@ -8,82 +8,78 @@ import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '.
...
-    } as any);
+  const onWorkloadChange = (selected: string) => {
+    setSelectedWorkloadUID(selected);
   };
 
-  onAddAsChange: React.ReactEventHandler<HTMLInputElement> = (event) => {
-    this.setState({
-      addAs: event.currentTarget.value,
-    });
+  // skipping handleChange
```
Comment (rhamilto) [inline frontend/public/components/modals/add-secret-to-workload.tsx:70]: It seems this no longer needed?  If so, remove it.

**Comment by tvu20** [comment]
This is technical debt cleanup and has no user-facing impact, so I am manually assigning docs and px approvals.

/assign @yapei for QE approval.
/label docs-approved
/label px-approved

**Comment by XiyunZhao** [comment]
This PR has been tested on private payload, and no issue was found    
/label qe-approved

**Pull request merged**